model/jmb_enest_nlparam.jl

```diff
@@ -0,0 +1,807 @@
+        - 
+# import demand                
+        sum((haskey(A.lookup[1], (r, g)) ? A[(r, g)] : 1.) * MD[r,g] for r in regions for g in goods if (a_set[r,g] != 0))
+);
+
+@mapping(cge,income_ra[r in regions],
+# consumption/utility
+        RA[r] 
+        - 
+        (
+# labor income        
+        PL[r] * sum(ld0_p[r,s] for s in sectors)
+# capital income        
+        + sum((haskey(PK.lookup[1], (r, s)) ? PK[(r,s)] : 1.0) * kd0_p[r,s] for s in sectors)
+# provision of household supply          
+        + sum( (haskey(PY.lookup[1], (r, g)) ? PY[(r, g)] : 1.0) * yh0_p[r,g] for g in goods)
+# revenue or costs of foreign exchange including household adjustment   
+        + PFX * (bopdef0_p[r] + hhadj_p[r])
+# government and investment provision        
+        - sum((haskey(PA.lookup[1], (r, g)) ? PA[(r, g)] : 1.0) * (g0_p[r,g] + i0_p[r,g]) for g in goods)
+# import taxes - assumes lumpsum recycling
+        + sum((haskey(A.lookup[1], (r, g)) ? A[(r, g)] : 1.) * MD[r,g] * PFX * tm_p[r,g] for g in goods if (a_set[r,g] != 0))
+# taxes on intermediate demand - assumes lumpsum recycling
+        + sum((haskey(A.lookup[1], (r, g)) ? A[(r, g)] : 1.) * a0_p[r,g]*(haskey(PA.lookup[1], (r, g)) ? PA[(r, g)] : 1.0)*ta_p[r,g] for g in goods if (a_set[r,g] != 0) )
+# production taxes - assumes lumpsum recycling  
+        + sum( (haskey(Y.lookup[1], (r, s)) ? Y[(r, s)] : 1) * ys0_p[r,s,g] * ty_p[r,s] for s in sectors, g in goods)
+        )
+);
+
+
+####################################
+# -- Complementarity Conditions --
+####################################
+
+# define complementarity conditions
+# note the pattern of ZPC -> primal variable  &  MCC -> dual variable (price)
+@complementarity(cge,profit_y,Y);
+@complementarity(cge,profit_x,X);
+@complementarity(cge,profit_a,A);
+@complementarity(cge,profit_c,C);
+@complementarity(cge,profit_ms,MS);
+@complementarity(cge,market_pa,PA);
+@complementarity(cge,market_py,PY);
+@complementarity(cge,market_pd,PD);
+@complementarity(cge,market_pn,PN);
+@complementarity(cge,market_pl,PL);
+@complementarity(cge,market_pk,PK);
+@complementarity(cge,market_pm,PM);
+@complementarity(cge,market_pc,PC);
+@complementarity(cge,market_pfx,PFX);
+@complementarity(cge,income_ra,RA);
+
+#energy nesting
+@complementarity(cge,profit_va,VA);
+@complementarity(cge,profit_e,E);
+@complementarity(cge,market_pva,PVA);
+@complementarity(cge,market_pe,PE);
+
+
+####################
+# -- Model Solve --
+####################
+
+#set up the options for the path solver
+PATHSolver.options(convergence_tolerance=1e-8, output=:yes, time_limit=3600)
+
+# export the path license string to the environment
+# this is now done in the SLiDE initiation steps 
+ENV["PATH_LICENSE_STRING"]="2617827524&Courtesy&&&USR&64785&11_12_2017&1000&PATH&GEN&31_12_2020&0_0_0&5000&0_0"
+
+# solve the model
+status = solveMCP(cge)
+
```

Comment:
### Module Integration
- [ ] For all of the model files, what output data are we interested in saving? What are your thoughts on how to tackle this? (see src/build/calibrate.jl)

Reply:
I haven't quite gotten here yet. Storing in dataframes maybe? I will look into calibrate.jl for ideas.